### PR TITLE
Adjusting to GraalVM for JDK21+

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Changes required to build on GraalVM for JDK21+ while keeping the support for building on GraalVM for JDK17.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
- All code has been tested:
  - [x] Tests still pass on JDK17

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Integrating these changes shall address few of the problems listed at 
- #7851

**Comment by JaroslavTulach** [comment]
There is a failure in [Mac Engine CI run](https://github.com/enso-org/enso/actions/runs/6249373260/job/16990810733?pr=7855#step:10:1342):
```
Could not find option with name js.foreign-object-prototype
```
that signals the GraalVM used in that run **doesn't have Graal.js installed**. How comes, @mwu-tow?

Shouldn't the `run backend build` command:
- download GraalVM
- install necessary components via `gu`
- only then start `sbt`

I remember having some discussion with you, @mwu-tow, and I think the goal was to avoid downloading anything in `sbt` - which is what [this PR does](https://github.com/enso-org/enso/pull/7855/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91L1316). 

I am surprised the failure happens only on Mac. E.g. on Linux the `./run` script probably downloads whatever needs to be downloaded. Any explanation?

Michal, if you know what to fix, please integrate your change into this PR's branch. Thank you in advance for any help.

**Comment by mwu-tow** [comment]
@JaroslavTulach It seems that JS was not marked as a required component, as it always came preinstalled with Graal distribution. I've pushed the change, let's see if it helps.

**Comment by JaroslavTulach** [comment]
Thank you Michal, everything is green. With an approval we can merge.

**Review by mwu-tow — APPROVED** [review]

**Review by Akirathan — APPROVED** [review]
I hope this change does not break any project-manager related runtime downloading. It should not.

**Review by hubertp — APPROVED** [review]

File: project/CopyTruffleJAR.scala [hunk 1]
```diff
@@ -83,6 +83,11 @@ object CopyTruffleJAR {
     libraryUpdates: UpdateReport,
     log: ManagedLogger
   ): Boolean = {
+    val version = System.getProperty("java.version").split("\\.");
+    if (Integer.parseInt(version(0)) >= 21) {
+      // skip all checks on GraalVM for JDK21 and newer
+      return false;
+    }
```
Comment (radeusgd) [inline project/CopyTruffleJAR.scala:90]: What is the reason for this change?

Is SBT now detecting the Truffle modules automatically since 21?

Comment (JaroslavTulach) [reply project/CopyTruffleJAR.scala:90]: Yes, Truffle API is "opened" for everyone since JDK21 because of https://medium.com/graalvm/truffle-unchained-13887b77b62c

Comment (radeusgd) [reply project/CopyTruffleJAR.scala:90]: Ok, I see.

So once we all upgrade to Graal 21, I assume we should be able to remove this line from `.jvmopts`:
```diff
 -Xss16M
 -Xmx4G
 -XX:+UseCompressedOops
---upgrade-module-path=engine/runtime/build-cache/truffle-api.jar
```
